File: app/src/main/java/wangdaye/com/geometricweather/common/basic/models/weather/WindDegree.java
package wangdaye.com.geometricweather.common.basic.models.weather;

import androidx.annotation.Nullable;

import java.io.Serializable;

/**
 * WindDegree.
 *
 * All properties are {@link androidx.annotation.NonNull}.
 * */
public class WindDegree implements Serializable {

    private final Float degree;
    private final Boolean noDirection;

    public WindDegree(Float degree, Boolean noDirection) {
        this.degree = degree;
        this.noDirection = noDirection;
    }

    public Float getDegree() {
        return degree;
    }

    public boolean isNoDirection() {
        return noDirection;
    }

    @Nullable
    public String getWindArrow() {
        if (noDirection) {
            return null;
        } else if (22.5 < degree && degree <= 67.5) {
            return "↙";
        } else if (67.5 < degree && degree <= 112.5) {
            return "←";
        } else if (112.5 < degree && degree <= 157.5) {
            return "↖";
        } else if (157.5 < degree && degree <= 202.5) {
            return "↑";
        } else if (202.5 < degree && degree <= 247.5) {
            return "↗";
        } else if (247.5 < degree && degree <= 292.5) {
            return "→";
        } else if (292. < degree && degree <= 337.5) {
            return "↘";
        } else {
            return "↓";
        }
    }
}

File: app/src/main/java/org/breezyweather/common/basic/models/weather/WindDegree.kt
package org.breezyweather.common.basic.models.weather

import java.io.Serializable

/**
 * WindDegree.
 */
class WindDegree(
    val degree: Float? = null,
    val isNoDirection: Boolean = false
) : Serializable {

    val windArrow: String?
        get() = if (degree == null) {
            null
        } else if (isNoDirection) {
            "⟳"
        } else when(degree) {
            in 22.5..67.5 -> "↙"
            in 67.5..112.5 -> "←"
            in 112.5..157.5 -> "↖"
            in 157.5..202.5 -> "↑"
            in 202.5..247.5 -> "↗"
            in 247.5..292.5 -> "→"
            in 292.5..337.5 -> "↘"
            else -> "↓"
        }
}
